tools/test.py

```diff
@@ -1505,7 +1534,7 @@ def Main():
 
   result = None
   def DoSkip(case):
-    return SKIP in case.outcomes or SLOW in case.outcomes
+    return SKIP in case.outcomes or SLOW in case.outcomes or (FLAKY in case.outcomes and options.flaky_tests == "skip")
```

Comment:
Unnecessary parens and very long line.  I'd break it up in separate if statements.  "skip" should really be SKIP.